ext/mbstring/oniguruma/src/regint.h

```diff
@@ -208,59 +238,81 @@
 #endif
 
 #define GET_ALIGNMENT_PAD_SIZE(addr,pad_size) do {\
-  (pad_size) = WORD_ALIGNMENT_SIZE \
-               - ((size_t)(addr) % WORD_ALIGNMENT_SIZE);\
+  (pad_size) = WORD_ALIGNMENT_SIZE - ((uintptr_t )(addr) % WORD_ALIGNMENT_SIZE);\
```

Comment:
This part of the patch should to be kept.

Reply:
The patch for 6.3.0 changed `(unsigned int)` to `(size_t)`; however, kkos/oniguruma@3ba00e1781c32468ef4acc683d791e0ffc36bc01 changed the `(unsigned int)` to `(uintptr_t)`. If the latter is not correct, it should be fixed in upstream as well.

Reply:
`uintptr_t` itself is correct, it normally should be same as `size_t`. It was earlier needed for 64-bit. The other part of it is, that `uintptr_t` a C99 thing. We use `intptr_t` and other types already, so it might be ok then. Anyway, no need for the upstream fixes, as long as there's no C89 and lower targeted.

Given the above, IMO we are fine to merge your patch as is then. The portability is ok for the platforms we know, and otherwise we can still turn the old patch part back on.

Thanks.

Reply:
Ah, I see. Thanks for the explanation, Anatol! :)

> Given the above, IMO we are fine to merge your patch as is then.

If there are no further objections, I'm going to apply the PR in a few days (unless somebody else beats me to it).